lab6/Client/src/Client/Common/DataChange.java

```diff
@@ -33,6 +32,8 @@ public static Object receiveObject() throws IOException {
     public static Response[] receiveAll() throws IOException {
         Response[] arg = new Response[20];
         try {
+            Thread.sleep(200);
+            if (connection.getSocket().getInputStream().available() <= 0) return null;
```

Comment:
Получается выполнение каждой команды замедлится как минимум на 200 милисекунд? Это неправильно.
И мне кажется это вообще не будет работать, ведь у тебя если сервер занят одним клиентом, то второй клиент повиснет еще при попытке подключения, разве нет?

Reply:
При подключении второго клиента все работает корректно. Как вариант, нельзя эти строчки заменить на это?
```suggestion
long time = new Date().getTime();
            while (connection.getSocket().getInputStream().available() <= 0) {
                if ((new Date().getTime()- time) > 500) return null;
            }
```

Reply:
Ладно, давай так